src/webgpu/api/operation/rendering/stencil.spec.ts

```diff
@@ -0,0 +1,232 @@
+                return vec4<f32>(params.color);
+            }`,
+        }),
+        entryPoint: 'main',
+      },
+      primitive: { topology: 'point-list' },
+      depthStencil,
+    });
+  }
+
+  createBindGroupForTest(layout: GPUBindGroupLayout, data: TypedArrayBufferView): GPUBindGroup {
+    return this.device.createBindGroup({
+      layout,
+      entries: [
+        {
+          binding: 0,
+          resource: {
+            buffer: this.makeBufferWithContents(data, GPUBufferUsage.UNIFORM),
+          },
+        },
+      ],
+    });
+  }
+}
+
+export const g = makeTestGroup(StencilTest);
+
+g.test('stencil_compare_func')
+  .desc(
+    `
+  Tests that stencil comparison functions with the stencil reference value works as expected.
+  `
+  )
+  .params(u =>
+    u //
+      .combineWithParams([
+        { stencilCompare: 'always', stencilRefValue: 0, expectedColor: kStencilColor },
+        { stencilCompare: 'always', stencilRefValue: 1, expectedColor: kStencilColor },
+        { stencilCompare: 'always', stencilRefValue: 2, expectedColor: kStencilColor },
+        { stencilCompare: 'equal', stencilRefValue: 0, expectedColor: kBaseColor },
+        { stencilCompare: 'equal', stencilRefValue: 1, expectedColor: kStencilColor },
+        { stencilCompare: 'equal', stencilRefValue: 2, expectedColor: kBaseColor },
+        { stencilCompare: 'greater', stencilRefValue: 0, expectedColor: kBaseColor },
+        { stencilCompare: 'greater', stencilRefValue: 1, expectedColor: kBaseColor },
+        { stencilCompare: 'greater', stencilRefValue: 2, expectedColor: kStencilColor },
+        { stencilCompare: 'greater-equal', stencilRefValue: 0, expectedColor: kBaseColor },
+        { stencilCompare: 'greater-equal', stencilRefValue: 1, expectedColor: kStencilColor },
+        { stencilCompare: 'greater-equal', stencilRefValue: 2, expectedColor: kStencilColor },
+        { stencilCompare: 'less', stencilRefValue: 0, expectedColor: kStencilColor },
+        { stencilCompare: 'less', stencilRefValue: 1, expectedColor: kBaseColor },
+        { stencilCompare: 'less', stencilRefValue: 2, expectedColor: kBaseColor },
+        { stencilCompare: 'less-equal', stencilRefValue: 0, expectedColor: kStencilColor },
+        { stencilCompare: 'less-equal', stencilRefValue: 1, expectedColor: kStencilColor },
+        { stencilCompare: 'less-equal', stencilRefValue: 2, expectedColor: kBaseColor },
+        { stencilCompare: 'never', stencilRefValue: 0, expectedColor: kBaseColor },
+        { stencilCompare: 'never', stencilRefValue: 1, expectedColor: kBaseColor },
+        { stencilCompare: 'never', stencilRefValue: 2, expectedColor: kBaseColor },
+        { stencilCompare: 'not-equal', stencilRefValue: 0, expectedColor: kStencilColor },
+        { stencilCompare: 'not-equal', stencilRefValue: 1, expectedColor: kBaseColor },
+        { stencilCompare: 'not-equal', stencilRefValue: 2, expectedColor: kStencilColor },
```

Comment:
nit: Make `expectedColor` a private param by renaming it to `_expectedColor` to shorten the test case path length.

Reply:
ok, I rename `expectedColor` to `_expectedColor`.